src/mesh/mesh.cpp

```diff
@@ -1182,4 +1175,44 @@ Mesh::GetLevelsAndLogicalLocationsFlat() const noexcept {
   return std::make_pair(levels, logicalLocations);
 }
 
+void Mesh::SetBCNames_(ParameterInput *pin) {
+  mesh_bc_names = {pin->GetOrAddString("parthenon/mesh", "ix1_bc", "outflow"),
+                   pin->GetOrAddString("parthenon/mesh", "ox1_bc", "outflow"),
+                   pin->GetOrAddString("parthenon/mesh", "ix2_bc", "outflow"),
+                   pin->GetOrAddString("parthenon/mesh", "ox2_bc", "outflow"),
+                   pin->GetOrAddString("parthenon/mesh", "ix3_bc", "outflow"),
+                   pin->GetOrAddString("parthenon/mesh", "ox3_bc", "outflow")};
+  // JMM: This is needed because not all BCs are necessarily
+  // implemented for swarms
+  auto maybe = [](const std::string &s) {
+    return ((s == "outflow") || (s == "periodic")) ? s : "outflow";
+  };
+  mesh_swarm_bc_names = {
```

Comment:
Cool.  I think this API is more flexible when needing to specify different boundary conditions for fields vs swarms.

Just for my understanding, can you help me work through the scenario that I want to specify user defined boundary conditions that are unique for swarms and fields.  I would do

```
parthenon/mesh/ix1_bc=custom_mesh_ix1
parthenon/swarm/ix1_bc=custom_swarm_ix1
```
with 
```
pman.app_input->RegisterBoundaryCondition(parthenon::BoundaryFace::inner_x1, "custom_mesh_ix1", MyMeshBoundaryInnerX1);
pman.app_input->RegisterSwarmBoundaryCondition(parthenon::BoundaryFace::inner_x1, "custom_swarm_ix1", MySwarmBoundaryInnerX1);
```
 
 or equivalently
 
 ```
parthenon/mesh/ix1_bc=user
```
with 
```
pman.app_input->RegisterBoundaryCondition(parthenon::BoundaryFace::inner_x1, MyMeshBoundaryInnerX1);
pman.app_input->RegisterSwarmBoundaryCondition(parthenon::BoundaryFace::inner_x1, MySwarmBoundaryInnerX1);
```

is that right?

Reply:
Yep that's right!